File: DSA/Java/tests/dsa/lib/math/nums/KthLargestTest.java
package dsa.lib.math.nums;

import org.junit.Test;

import static org.junit.Assert.*;

public class KthLargestTest {
    public static class AddTest {
        @Test
        public void case1() {
            KthLargest kl = new KthLargest(3, new int[]{4, 5, 8, 2});
            assertEquals(4, kl.add(3));
            assertEquals(5, kl.add(5));
            assertEquals(5, kl.add(10));
            assertEquals(8, kl.add(9));
            assertEquals(8, kl.add(4));
        }
    }

}

File: DSA/Cpp/test/dsa/lib/math/nums/kth_largest_test.cpp
#include <dsa/lib/math/nums/KthLargest.hpp>
#include "nums_test.h"

namespace dsa::lib::math::nums {
namespace {

class KthLargestTest : public NumsTest {};
class AddTest : public KthLargestTest {};
TEST_F(AddTest, case1) {
    std::vector<int> nums = {4, 5, 8, 2};
    KthLargest kl(3, nums);
    EXPECT_EQ(4, kl.add(3));
    EXPECT_EQ(5, kl.add(5));
    EXPECT_EQ(5, kl.add(10));
    EXPECT_EQ(8, kl.add(9));
    EXPECT_EQ(8, kl.add(4));
}

}
}
